RestaurantsNearMe/Data/YelpRequest/RestaurantListProvider.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  RestaurantListProvider.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2024-01-20.
+//
+
+import Foundation
+import Networking
+import Combine
+
+struct RestaurantListProvider: RestaurantListProving {
```

Comment:
This is the boundary between presentation and `Networking`. I tried to follow your advice and abstract as many details as possible from ViewModels (i.e `SplashScreen<SomeNetworkingProtocol>`)

I distilled the bulk of the changes in this image:

<img width="695" alt="Screenshot 2024-01-20 at 11 05 29 PM" src="https://github.com/ColinEvans/RestaurantsNearMe/assets/9354994/0b859eee-c704-4e09-b454-efadb2ff4a49">